lib/spack/spack/build_systems/intel.py

```diff
@@ -994,6 +994,16 @@ def libs(self):
                 libnames,
                 root=self.component_lib_dir('mpi'),
                 shared=True, recursive=True) + result
+            # Intel MPI since 2019 depends on libfabric which is not in the
+            # lib directory but in a directory of its own which should be
+            # included in the rpath
+            if self.version_yearlike >= ver('2019'):
+                # Patch libmpi.so rpath so it can find libfabric
+                d = ancestor(self.component_lib_dir('mpi'))
+                libfabric_rpath = os.path.join(d, 'libfabric', 'lib')
```

Comment:
```suggestion
                libfabric_rpath = [os.path.join(d, 'libfabric', 'lib')]
```
later code assumes this is a list